srcpkgs/smpq/template

```diff
@@ -0,0 +1,19 @@
+# Template file for 'smpq'
+pkgname=smpq
+version=1.6
+revision=1
+build_style=cmake
+configure_args="-DWITH_KDE=OFF"
+hostmakedepends="StormLib"
```

Comment:
Why is this needed?  There's nothing I see in StormLib that would be needed on the host.

Reply:
smpq depends on StormLib to create the MPQ file, version devillutionX 1.2.1 everything was built-in, now it's not.

Reply:
Found with "hostmakedepends", which "depends", by cmake.

Reply:
If it depends on StormLib, StormLib-devel should go in makedepends; if its doing anything with the .so in hostmakedepends, then its wrong.  StormLib should be detected in depends as long as the entry in common/shlibs is correct and it links correctly.

Reply:
I changed "hostmakedepends" to "depends", this error message appears on compilation:

CMake Error at CMakeLists.txt:43 (message):
  StormLib or storm library not found


-- Configuring incomplete, errors occurred!
See also "/builddir/smpq-1.6/build/CMakeFiles/CMakeOutput.log".
=> ERROR: smpq-1.6_1: do_configure: 'CFLAGS="-DNDEBUG ${CFLAGS/ -pipe / }" CXXFLAGS="-DNDEBUG ${CXXFLAGS/ -pipe / }" cmake ${cmake_args} ${configure_args} ${LIBS:+-DCMAKE_C_STANDARD_LIBRARIES="$LIBS"} ${LIBS:+-DCMAKE_CXX_STANDARD_LIBRARIES="$LIBS"} ${wrksrc}/${build_wrksrc}' exited with 1                                           
=> ERROR:   in do_configure() at common/build-style/cmake.sh:75

Reply:
It shouldn't go in `depends`, either, that'll be automatically detected.  I removed the `hostmakedepends` line and the `do_build()` function (since it doesn't do anything).  `smpq` built fine.

Reply:
I can't recreate the issue to which you're referring; I removed the `hostmakedepends` line and the pointless `do_build()` and it builds just fine.

```
% cat srcpkgs/smpq/template 
# Template file for 'smpq'
pkgname=smpq
version=1.6
revision=1
build_style=cmake
configure_args="-DWITH_KDE=OFF"
makedepends="bzip2-devel zlib-devel StormLib-devel"
short_desc="StormLib MPQ archiving utility"
maintainer="MarcoAPC <marcoaureliopc@gmail.com>"
license="GPL-3.0-or-later"
homepage="https://launchpad.net/smpq"
distfiles="https://launchpad.net/smpq/trunk/${version}/+download/smpq_${version}.orig.tar.gz"
checksum=b5d2dc8a5de8629b71ee5d3612b6e84d88418b86c5cd39ba315e9eb0462f18cb
```

```
=> Registering new packages to /host/binpkgs/patch-9
index: added `smpq-1.6_1' (aarch64).
```

Reply:
The problem was related to StormLib version 9.23, I updated to 9.24 and removed the lines from the smpq template.

Reply:
Done!